doc/vrrp/VRRP_Adaptation_HLD.md

```diff
@@ -0,0 +1,712 @@
+  - Match VRRP instance tracking interface and recalculate priority
+    - If track interface goes down, the priority of VRRP instance will be reduced by weight
+    - If track interface goes up, the priority of VRRP instance will be added by weight
+  - Update priority change by using FRR CLI commands
+
+vrrpd:
+  - Update VRRP packet priority field
+  - Send VRRP packet to Linux Network stack
+
+### SAI API 
+
+VRRP virtual router interface can be created by setting the attribute “SAI_ROUTER_INTERFACE_ATTR_IS_VIRTUAL” with create_router_interface API.
+```
+ * @brief RIF creation is a virtual RIF.
+ *
+ * Create a Virtual RIF object, which only programs the ingress router MAC.
+ * This simplifies the management of VRRP master router's configuration in
+ * SAI adapter, as defined by RFC 5798 (or similar proprietary protocols).
+ * Using a Virtual RIF allows SAI to optimize resources, so neighbor entries
+ * cannot be learned on a Virtual RIF. On a virtual RIF following attributes
+ * are invalid: ADMIN state, MTU size, packet action and multicast enable.
+ * Alternatively VRRP can also be configured using native RIF objects without
+ * using VIRTUAL attribute, with the expectation that SAI adapter will consume
+ * resources that will not be used.
+ *
+ * @type bool
+ * @flags CREATE_ONLY
+ * @default false
+ */
+
+SAI_ROUTER_INTERFACE_ATTR_IS_VIRTUAL,
+
+```
+
+### Configuration and management 
+
+#### Manifest
+
+N/A
+
+#### CLI/YANG Model Enhancements
+
+##### SNOiC VRRP YANG MODEL
+
+```yang
+module sonic-vrrp {
+
+    namespace "http://github.com/Azure/sonic-vrrp";
+    prefix vrrp;
+
+    import ietf-inet-types {
+        prefix inet;
+    }
+
+    organization
+        "SONiC";
+
+    contact
+        "SONiC";
+
+    description "VRRP yang Module for SONiC OS";
+
+    revision 2023-08-10 {
+        description "add sonic-vrrp.yang";
+    }
+
+    container sonic-vrrp {
+
+        container VRRP {
+
+            description "VRRP part of configdb.json"
+
+            list VRRP_LIST {
+
+                key "interface_name vrid";
+
+                leaf interface_name {
+                    type union {
+                        type string {
+                            pattern "Ethernet([1-3][0-9]{3}|[1-9][0-9]{2}|[1-9][0-9]|[0-9])";
+                        }
+                        type string {
+                            pattern "PortChannel[0-9]{1,4}";
+                        }
+                        type string {
+                            pattern "Vlan(409[0-4]|40[0-8][0-9]|[1-3][0-9]{3}|[1-9][0-9]{2}|[1-9][0-9]|[1-9])";
+                        }
+                    } 
+                }
+
+                leaf vrid {
+                    type uint8 {
+                        range 1..255;
+                    }
+                }
+
+                leaf-list vip {
+                    type union {
+                        type inet:ipv4-prefix;
+                        type inet:ipv6-prefix;
```

Comment:
Why are we using prefix instead of inet:ip-address?

Reply:
There will some issue with prefix, because prefix will carry no mask, the 32-bit mask default routing causes ping failure issue.